lib/Alchemy/Phrasea/Core/Event/Subscriber/ElasticsearchSubscriber.php

```diff
@@ -0,0 +1,194 @@
+<?php
+
+/*
+ * This file is part of Phraseanet
+ *
+ * (c) 2005-2014 Alchemy
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Alchemy\Phrasea\Core\Event\Subscriber;
+
+use Alchemy\Phrasea\Core\Event\CollectionEvent\CollectionIndexEvent;
+use Alchemy\Phrasea\Core\Event\DataboxEvent\DataboxIndexEvent;
+use Alchemy\Phrasea\Core\Event\DataboxEvent\IndexEvent;
+use Alchemy\Phrasea\Core\Event\RecordEvent\RecordIndexEvent;
+use Alchemy\Phrasea\Core\PhraseaEvents;
+use Alchemy\Phrasea\Core\PhraseaTokens;
+use Alchemy\Phrasea\SearchEngine\Elastic\BulkOperation;
+use Alchemy\Phrasea\SearchEngine\Elastic\Fetcher\RecordFetcher;
+use Alchemy\Phrasea\SearchEngine\Elastic\Fetcher\RecordPoolFetcher;
+use Alchemy\Phrasea\SearchEngine\Elastic\Indexer\RecordIndexer;
+use Alchemy\Phrasea\SearchEngine\Elastic\Indexer\TermIndexer;
+use Alchemy\Phrasea\SearchEngine\Elastic\RecordHelper;
+use Elasticsearch\Client;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+class ElasticSearchSubscriber implements EventSubscriberInterface
```

Comment:
It's the responsibility of the indexer component to do many of the work done in this class. A subscriber should be just about doing some plumbing work.
As a proof of matter, this class is not easily testable. It's not possible to check the business logic without magic-dancing with the event dispatcher. It's a little code smell sign IMO.